### PR TITLE
[ISSUE 3932] refactor MotanServiceEventListener

### PR DESCRIPTION
**Pull request by misaya295**
about #3932 

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-client/shenyu-client-motan/src/main/java/org/apache/shenyu/client/motan/MotanServiceEventListener.java [hunk 1]
```diff
@@ -89,52 +88,52 @@ public MotanServiceEventListener(final PropertiesConfig clientConfig, final Shen
...
+
     }
 
     @Override
     public void onApplicationEvent(final ContextRefreshedEvent contextRefreshedEvent) throws BeansException {
         applicationContext = contextRefreshedEvent.getApplicationContext();
         Map<String, Object> beans = applicationContext.getBeansWithAnnotation(ShenyuMotanClient.class);
         for (Map.Entry<String, Object> entry : beans.entrySet()) {
-            handler(entry.getValue());
+            handle("ShenyuMotanClient", entry.getValue());
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-motan/src/main/java/org/apache/shenyu/client/motan/MotanServiceEventListener.java:99]: unnecessary to override `#onApplicationEvent`.

File: shenyu-client/shenyu-client-motan/src/main/java/org/apache/shenyu/client/motan/MotanServiceEventListener.java [hunk 2]
```diff
@@ -167,11 +166,45 @@ private MetaDataRegisterDTO buildMetaDataDTO(final Class<?> clazz, final MotanSe
...
+                : pathJoin(this.contextPath, superPath, methodShenyuClient.path());
+    }
+
+    @Override
+    protected void handle(final String beanName, final Object bean) {
+        if (group == null) {
+            group = ((BasicServiceConfigBean) applicationContext.getBean(BASE_SERVICE_CONFIG)).getGroup();
+        }
+        Class<?> clazz = getCorrectedClass(bean);
+        MotanService service = clazz.getAnnotation(MotanService.class);
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-motan/src/main/java/org/apache/shenyu/client/motan/MotanServiceEventListener.java:187]: `service` not used here, why rewrite `#handle`?

File: shenyu-client/shenyu-client-motan/src/main/java/org/apache/shenyu/client/motan/MotanServiceEventListener.java [hunk 3]
```diff
@@ -89,52 +88,52 @@ public MotanServiceEventListener(final PropertiesConfig clientConfig, final Shen
...
-        String path = superPath.contains("*") ? pathJoin(contextPath, superPath.replace("*", ""), method.getName()) : pathJoin(contextPath, superPath, shenyuMotanClient.path());
+    @Override
+    protected Class<ShenyuMotanClient> getAnnotationType() {
+        return ShenyuMotanClient.class;
+    }
+
+    @Override
+    protected MetaDataRegisterDTO buildMetaDataDTO(final Object bean, final ShenyuMotanClient shenyuMotanClient, final String superPath, final Class<?> clazz, final Method method) {
+        Integer timeout = Optional.ofNullable(((BasicServiceConfigBean) applicationContext.getBean(BASE_SERVICE_CONFIG)).getRequestTimeout()).orElse(1000);
+        MotanService service = clazz.getAnnotation(MotanService.class);
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-motan/src/main/java/org/apache/shenyu/client/motan/MotanServiceEventListener.java:135]: use `AnnotatedElementUtils#findMergedAnnotation` to replace `Class#getAnnotation` for better scalability.

**Comment by misaya295** [comment]
@dragon-zhang Hello zhang, I have solved some comments, pls take some time to review,thk u.

File: shenyu-client/shenyu-client-motan/src/main/java/org/apache/shenyu/client/motan/MotanServiceEventListener.java (hunk 3 above)
Comment (dragon-zhang) [reply shenyu-client/shenyu-client-motan/src/main/java/org/apache/shenyu/client/motan/MotanServiceEventListener.java]: Can you give a reason for using `AnnotatedElementUtils#getMergedAnnotation`?

Comment (misaya295) [reply shenyu-client/shenyu-client-motan/src/main/java/org/apache/shenyu/client/motan/MotanServiceEventListener.java]: Should I use AnnotatedElementUtils#findMergedAnnotation to replace Class#getAnnotation for better scalability?

Comment (misaya295) [reply shenyu-client/shenyu-client-motan/src/main/java/org/apache/shenyu/client/motan/MotanServiceEventListener.java]: > Can you give a reason for using `AnnotatedElementUtils#getMergedAnnotation`?

Oh,I got it, I should use findMergedAnnotation... I use wrong function..

**Pull request merged**